File: src/main/java/cn/labzen/logger/spring/LabzenLoggerInitializer.java
package cn.labzen.logger.spring;

import cn.labzen.logger.print.MetaPrinter;
import cn.labzen.meta.spring.SpringInitializationOrder;
import org.springframework.context.ApplicationContextInitializer;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.core.Ordered;

import javax.annotation.Nonnull;

/**
 * 在Spring Boot的项目启动时执行
 */
public class LabzenLoggerInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>, Ordered {

  @Override
  public int getOrder() {
    return SpringInitializationOrder.MODULE_LOGGER_INITIALIZER_ORDER;
  }

  @Override
  public void initialize(@Nonnull ConfigurableApplicationContext applicationContext) {
    MetaPrinter.print();
  }
}

File: src/main/kotlin/cn/labzen/logger/spring/LabzenLoggerInitializer.kt
package cn.labzen.logger.spring

import cn.labzen.logger.meta.LabzenMetaPrinter
import cn.labzen.logger.meta.LoggerConfiguration
import cn.labzen.meta.Labzens
import cn.labzen.meta.spring.SpringApplicationContextInitializerOrder
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.core.Ordered

/**
 * 在Spring Boot的项目启动时执行
 */
class LabzenLoggerInitializer : ApplicationContextInitializer<ConfigurableApplicationContext>, Ordered {

  override fun getOrder(): Int =
    SpringApplicationContextInitializerOrder.MODULE_LOGGER_INITIALIZER_ORDER

  override fun initialize(applicationContext: ConfigurableApplicationContext) {
    val configuration = Labzens.configurationWith(LoggerConfiguration::class.java)

    // 打印 Labzen 模组信息
    if (configuration.printMetaInformation()) {
      LabzenMetaPrinter.print()
    }
  }
}
